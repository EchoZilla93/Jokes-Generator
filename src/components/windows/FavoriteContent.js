import React from 'react';
import {withGetScreen} from 'react-getscreen';

import FavoriteCardBluePrint from '../Favorite/FavoriteCardBluePrint';
import './favoriteStyles.scss'
import FavTablet from '../favTablet/FavTablet';

export const FavoriteContent = props =>{
    if (props.isTablet()) return <FavTablet/>;
    if(props.isMobile()) return <FavTablet/>;
    return(
        <aside className="favorite">
            <h1>Favorite</h1>
            <FavoriteCardBluePrint />
        </aside>
    )
}

const options = {mobileLimit: 500, tabletLimit: 1025}
export default withGetScreen(FavoriteContent,options);