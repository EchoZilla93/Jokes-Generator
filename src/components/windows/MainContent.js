import React from 'react';

import Head from '../Header/Head';
import CheckBoxesRender from '../Checkbox/CheckBoxesRender';
import GetJokeBtn from '../../components/GetJokeBtn/GetJokeBtn';

export const MainContent = () => {
    return (
        <div>
            <Head/>
            <CheckBoxesRender/>
            <GetJokeBtn/>
        </div>
    )
}

export default MainContent;