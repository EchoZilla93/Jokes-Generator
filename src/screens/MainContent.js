import React from 'react';

import Head from '../components/Head';
import CheckBoxSelectors  from '../components/CategoryBoxSelectors';
import RandomBoxSelection from '../components/RandomBoxSelector';
import Search from '../components/Search';
import GetJokeBtn from '../components/GetJokeBtn';
import JokeCard from '../components/JokeCard';

const MainContent = () => {
    return (
        <div>
            <Head/>
            <RandomBoxSelection/>
            <CheckBoxSelectors />
            <Search/>
            <GetJokeBtn/>
        </div>
    )
}

export default MainContent;