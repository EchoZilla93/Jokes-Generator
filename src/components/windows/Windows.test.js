import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import MainContent from './MainContent';
import FavoriteContent from './FavoriteContent';

describe('windows content',() => {
    it('mainwindow renders expect snapshot',() => {
        const wrapper = shallow(<MainContent/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('favoritewindow renders expect snapshot',() => {
        const wrapper = shallow(<FavoriteContent/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})