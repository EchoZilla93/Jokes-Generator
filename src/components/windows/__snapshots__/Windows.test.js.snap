// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`windows content favoritewindow renders expect snapshot 1`] = `
<FavoriteContent
  isDesktop={[Function]}
  isMobile={[Function]}
  isTablet={[Function]}
/>
`;

exports[`windows content mainwindow renders expect snapshot 1`] = `
<div>
  <Head />
  <CheckBoxRender />
  <GetJokeBtn />
</div>
`;
